AsyncDisplayKit/Layout/ASLayoutablePrivate.h

```diff
@@ -60,9 +59,202 @@ extern void ASLayoutableClearCurrentContext();
+}\
+\
+- (void)setFlexGrow:(BOOL)flexGrow\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.flexGrow = flexGrow;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (BOOL)flexShrink\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.flexShrink;\
+}\
+\
+- (void)setFlexShrink:(BOOL)flexShrink\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.flexShrink = flexShrink;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (ASRelativeDimension)flexBasis\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.flexBasis;\
+}\
+\
+- (void)setFlexBasis:(ASRelativeDimension)flexBasis\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.flexBasis = flexBasis;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (ASStackLayoutAlignSelf)alignSelf\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.alignSelf;\
+}\
+\
+- (void)setAlignSelf:(ASStackLayoutAlignSelf)alignSelf\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.alignSelf = alignSelf;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (CGFloat)ascender\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.ascender;\
+}\
+\
+- (void)setAscender:(CGFloat)ascender\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.ascender = ascender;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (CGFloat)descender\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.descender;\
+}\
+\
+- (void)setDescender:(CGFloat)descender\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.descender = descender;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (ASRelativeSizeRange)sizeRange\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.sizeRange;\
+}\
+\
+- (void)setSizeRange:(ASRelativeSizeRange)sizeRange\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.sizeRange = sizeRange;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+\
+- (CGPoint)layoutPosition\
+{\
+  ASDN::MutexLocker l(_propertyLock);\
+  return _environmentState.layoutOptionsState.layoutPosition;\
+}\
+\
+- (void)setLayoutPosition:(CGPoint)layoutPosition\
+{\
+  _propertyLock.lock();\
+  _environmentState.layoutOptionsState.layoutPosition = layoutPosition;\
+  [self propagateUpLayoutOptionsState];\
+  _propertyLock.unlock();\
+}\
+
+
+#pragma mark - ASLayoutableExtensibility
+
+#define ASEnvironmentLayoutExtensibilityForwarding \
+- (void)setLayoutOptionExtensionBool:(BOOL)value atIndex:(int)idx\
+{\
+  _ASEnvironmentLayoutOptionsExtensionSetBoolAtIndex(self, idx, value);\
```

Comment:
We may need locking these methods, I don't have time to confirm right now, but can analyze more closely tomorrow.